### PR TITLE
Implemente eos-evm gas-fee (G_codedeposit and G_txnewaccount)

### PR DESCRIPTION
**Pull request by elmato**

### CODE REVIEW

**Review by yarkinwho — COMMENTED** [review]
Seems fine, but one question: where should we put the gas_parameters struct?

I have a similar structure in silkworm as well, we should unify them if possible.

**Review by yarkinwho — APPROVED** [review]
Looks fine. remember to test the trace result later.

File: silkworm/node/stagedsync/stages/stage_execution.cpp [hunk 1]
```diff
@@ -627,5 +627,13 @@ void Execution::unwind_state_from_changeset(db::ROCursor& source_changeset, db::
         revert_state(new_key, new_value, plain_state_table, plain_code_table);
         src_data = source_changeset.to_previous(/*throw_notfound*/ false);
     }
+
+}
+
+evmone::gas_parameters Execution::get_gas_params(const Block& block) const{
```
[the remark this answers is not in the document]
Comment (elmato) [reply silkworm/node/stagedsync/stages/stage_execution.cpp:633]: Yes, we have the tx passed as argument in the forward function. I will do that in the other PR

File: silkworm/core/chain/config.cpp [hunk 2]
```diff
@@ -196,13 +196,20 @@ evmc_revision ChainConfig::revision(const BlockHeader& header) const noexcept {
...
+    return eosevm::version_to_evmc_revision(evm_version);
+}
+
+uint64_t ChainConfig::eos_evm_version(const BlockHeader& header) const noexcept {
     uint64_t evm_version = 0;
-    if(header.number == 0) {
-        evm_version = _version.has_value() ? *_version : 0;
-    } else {
-        evm_version = eosevm::nonce_to_version(header.nonce);
+    if(protocol_rule_set == protocol::RuleSetType::kTrust) {
```
[the remark this answers is not in the document]
Comment (elmato) [reply silkworm/core/chain/config.cpp:205]: We wanted to preserve the old behavior in order to pass the original tests

**Pull request merged**